src/adapter/http/src/simple_protocol/handlers.rs

```diff
@@ -273,4 +280,8 @@ fn get_base_dataset_url(
     base_url_rest.join(path_string.as_str()).unwrap()
 }
 
+fn ensure_version_compatibility(client_version: i32) -> bool {
```

Comment:
Could return `Result<(), ApiError>` instead of `bool`.

Also consider making this a method of `OdfSmtpVersion` header - this will make it easier to find.